index/scorch/scorch.go

```diff
@@ -765,3 +787,53 @@ func parseToInteger(i interface{}) (int, error) {
 		return 0, fmt.Errorf("expects int or float64 value")
 	}
 }
+
+type fieldStats struct {
+	statMap map[string]map[string]uint64
```

Comment:
Would this map access not need a lock, or are you controlling the access to it such that you'd never call read until it's completely updated?

Reply:
Add commentary around what this map's content is.

Reply:
The map shouldn't require a lock. Data is written into the map immediately after making the zap segment and remains unchanged throughout the segment's lifetime (As of now, adding additional stats may change this behaviour). There isn't a code path that accesses the map before it is written into or while it is being populated, so I believe this should be fine.

I will address the commentary though